WcaOnRails/app/controllers/api/internal/v1/users_controller.rb

```diff
@@ -5,9 +5,13 @@ class Api::Internal::V1::UsersController < Api::Internal::V1::ApiController
   protect_from_forgery except: [:competitor_info]
   def competitor_info
     competitors = params.require(:ids)
-    users = User.find_all(competitors)
-    render json: users.to_json({
-                                 only: %w[id wca_id name gender country_iso2 email dob],
-                               })
+    users = User.find(competitors)
+    data = users.map { |u|
+      u.serializable_hash({
+                            only: %w[id wca_id name gender country_iso2 email dob],
+                          },
+                          overwrite_default: true)
+    }
```

Comment:
(If GitHub renders this comment at the top of the list, read my other comment first)

Please retain the old `to_json` invocation. The merge between the `DEFAULT_SERIALIZE_OPTIONS` and your custom options is **intentionally not** recursive. If you don't care about certain fields, you can skip them by providing an empty array.

Reply:
thanks! I was trying to find a solution like this!